pkg/check/feed/feed.go

```diff
@@ -0,0 +1,174 @@
+}
+
+// NewCheck returns a new check instance.
+func NewCheck(logger logging.Logger) beekeeper.Action {
+	return &Check{
+		logger: logger,
+	}
+}
+
+func (c *Check) Run(ctx context.Context, cluster orchestration.Cluster, opts interface{}) (err error) {
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	if o.RootRef != "" {
+		c.logger.Infof("running availability check")
+		return c.availability(ctx, cluster, o)
+	}
+	return c.regular(ctx, cluster, o)
+}
+
+func (c *Check) availability(ctx context.Context, cluster orchestration.Cluster, o Options) error {
+	ref, err := swarm.ParseHexAddress(o.RootRef)
+	if err != nil {
+		return fmt.Errorf("invalid root ref: %w", err)
+	}
+
+	nodeNames := cluster.FullNodeNames()
+	nodeName := nodeNames[0]
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+
+	client := clients[nodeName]
+	_, _, err = client.DownloadFile(ctx, ref, nil)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Check) regular(ctx context.Context, cluster orchestration.Cluster, o Options) error {
+	nodeNames := cluster.FullNodeNames()
+	nodeName := nodeNames[0]
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+	upClient := clients[nodeName]
+
+	c.logger.Infof("upload client: %s", upClient.Name())
+
+	batchID, err := upClient.GetOrCreateMutableBatch(ctx, o.PostageAmount, o.PostageDepth, o.PostageLabel)
+	if err != nil {
+		return err
+	}
+
+	privKey, err := crypto.GenerateSecp256k1Key()
+	if err != nil {
+		return err
+	}
+
+	signer := crypto.NewDefaultSigner(privKey)
+	topic, err := crypto.LegacyKeccak256([]byte("my-topic"))
+	if err != nil {
+		return err
+	}
+
+	// create root
+	createManifestRes, err := upClient.CreateRootFeedManifest(ctx, signer, topic, api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return err
+	}
+	c.logger.Infof("node %s: manifest created", nodeName)
+	c.logger.Infof("reference: %s", createManifestRes.Reference)
+	c.logger.Infof("owner: %s", createManifestRes.Owner)
+	c.logger.Infof("topic: %s", createManifestRes.Topic)
+
+	// make updates
+	for i := 0; i < o.NUpdates; i++ {
+		time.Sleep(3 * time.Second)
+		data := fmt.Sprintf("update-%d", i)
+		fName := fmt.Sprintf("file-%d", i)
+		file := bee.NewBufferFile(fName, bytes.NewBuffer([]byte(data)))
+		err = upClient.UploadFile(context.Background(), &file, api.UploadOptions{
+			BatchID: batchID,
+			Direct:  true,
+		})
+		if err != nil {
+			return err
+		}
+		ref := file.Address()
+		socRes, err := upClient.UpdateFeedWithReference(ctx, signer, topic, uint64(i), ref, api.UploadOptions{BatchID: batchID})
+		if err != nil {
+			return err
+		}
+		c.logger.Infof("node %s: feed updated", nodeName)
+		c.logger.Infof("soc reference: %s", socRes.Reference)
+		c.logger.Infof("wrapped reference: %s", file.Address())
+	}
+	time.Sleep(5 * time.Second)
+
+	// fetch update
+	c.logger.Infof("fetching feed update")
+	downClient := clients[nodeNames[1]]
```

Comment:
client selection should ideally be random...also if the slice size is less than 2, this could lead to a panic.